llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -1764,6 +1764,19 @@ Instruction *InstCombinerImpl::visitFDiv(BinaryOperator &I) {
     return replaceInstUsesWith(I, Pow);
   }
 
+  // powi(X, Y) / X --> powi(X, Y-1)
+  // Todo: Multi-use may be also better off creating Powi(x,y-1)
+  if (I.hasNoInfs() && I.isFast() &&
```

Comment:
Don't use isFast(), check specific required flags.

Reply:
Thank you for your suggestion, and I want to make sure that you're referring to I.getFastMathFlags().isFast() ?

Reply:
Thanks for your  detailed explanation. Now adjust it with **I.hasNoInfs() && I.hasAllowReassoc()**